caster-back/story_graph/engine.py

```diff
@@ -311,6 +323,79 @@ async def execute_node(
             else:
                 log.error(f"Occured invalid/unknown CellType {cell_type}")
 
+    async def _evaluate_python_code(self, code: str) -> bool:
+        stream_variables = await self.get_stream_variables()
+        try:
+            r = eval(
+                code,
+                self.get_engine_global_vars(
+                    {
+                        "loop": asyncio.get_event_loop(),
+                        "vars": stream_variables,
+                        "self": self,
+                        "get_stream_variables": self.get_stream_variables,
+                        "wait_for_stream_variable": self.wait_for_stream_variable,
+                    }
+                ),
+            )
+        except Exception:
+            raise InvalidPythonCode()
+        if not isinstance(r, bool):
+            log.debug(f"Return type of '{code}' is not a boolean but {type(r)}")
+            raise InvalidPythonCode()
+        return r
+
+    async def get_next_node(self) -> Node:
+        """Iterates over each exit :class:`~NodeDoor`
+        of the current node and evaluates its boolean value
+        and decides.
+
+        If the node door code consists of invalid code it will be skipped.
+        If all boolean evaluations result in ``False`` or invalid code,
+        the default exit will be used.
+
+        If multiple out-going edges are connected to an active door,
+        a random edge will be picked to follow for the next node.
+
+        If the node does not have any out-going edges a :class:`~GraphDeadEnd`
```

Comment:
This should trigger the end-screen, right?

Reply:
It should :tm:
But see  https://github.com/Gencaster/gencaster/issues/530#issuecomment-1712911247